### PR TITLE
Add support for ngModel

### LINKED ISSUE #5: value and ngModel
Hi, I have a problem when I am using td-code-editor inside a form, using "value" property the assigned text is displayed on editor but I can't save the changes.

```
<td-code-editor style="height: 200px"
            id="statement"
            placeholder="Statement"
            required
            [value]="query.statement"   
            name="statement"
            theme="vs"
            flex
            language="sql">
          </td-code-editor>
```
 Using ngModel I can't see the assigned text, but I can save anything new I type on the editor. 
```
<td-code-editor style="height: 200px"
            id="statement"
            placeholder="Statement"
            required
            [ngModel]="query.statement"   // or [(ngModel)]
            ngDefaultControl
            name="statement"
            theme="vs"
            flex
            language="sql">
          </td-code-editor>
```
I tried several ways to connect value with the ngModel or ngModel with $event, then I was able to display the assigned text and save the changes, but every time I am typing a character at the end or in the middle of a previous text, after the first character the input point is going automatically to the beginning of the text. Also tried to use the onEditorValueChange event without results. So how can I use the editor and save the changes? Thank you

### PR DESCRIPTION
**Pull request by jeremysmartt**
## Description

Adding support for ngModel
fixes #5 

### What's included?

- modified:   docs/API.md
- modified:   package.json
- modified:   src/code-editor/code-editor.component.ts

#### Test Steps

- [ ] `npm run test` passes
- [ ] `ng lint` passes

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -26,6 +27,7 @@
 | `registerLanguage` | `function()` | Registers a custom Language within the editor
 | `editorStyle` | `string` | Additional Styling applied to Editor Container
 | `theme` | `string` | Theme used to style the Editor
-| `isElectronApp` | `function()` | Returns true or false based on if running in Electron.
+| `isElectronApp` | `function()` | Returns true or false based on if running in Electron
+| `[(ngModel)]` | `directive` | One-way or two-way binding of the variable value provided
```
Comment (emoralesb05) [inline docs/API.md]: Typically there is no need to add this as a property, we can just state that we have forms support.

File: docs/API.md [hunk 2]
```diff
@@ -6,18 +6,19 @@
...
 
-#### The <td-code-editor> component has 4 events:
+#### The <td-code-editor> component has 5 events:
 
 | Name | Description |
 | --- | --- |
 | `onEditorInitialized` | Emitted when Editor is finished initializing
 | `onEditorConfigurationChanged` | Emitted when configuration of the Editor changes
 | `onEditorLanguageChanged` | Emitted when the language of the Editor changes
 | `onEditorValueChange` | Emitted when value of the Editor changes
```
Comment (emoralesb05) [inline docs/API.md]: would this be needed if we use `(change)` and `ngModel`?

**Review by emoralesb05 — CHANGES_REQUESTED** [review]

File: docs/API.md (hunk 2 above)
Comment (jeremysmartt) [reply docs/API.md]: i left it because studio is using it.  Want to talk with them before deprecating.

Comment (emoralesb05) [reply docs/API.md]: You can deprecate it, and remove it from docs without removing the implementation

File: docs/API.md (hunk 1 above)
Comment (jeremysmartt) [reply docs/API.md]: done

File: docs/API.md (hunk 2 above)
Comment (jeremysmartt) [reply docs/API.md]: done

**Review by emoralesb05 — APPROVED** [review]

**Pull request merged**